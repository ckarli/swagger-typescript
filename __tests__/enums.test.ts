import { generator } from "../src/generator";
import swaggerJson from "./swagger.json";

describe("generate", () => {
  //@ts-ignore
  const { code, hooks, type } = generator(swaggerJson, {
    generateEnumAsType: true,
  });

  it("generate Code", () => {
    expect(code).toMatchSnapshot();
  });
  it("generate hooks", () => {
    expect(hooks).toMatchSnapshot();
  });
  it("generate type", () => {
    expect(type).toMatchSnapshot();
  });
});
