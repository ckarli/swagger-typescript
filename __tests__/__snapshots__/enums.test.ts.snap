// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate generate Code 1`] = `
"
/**
* AUTO_GENERATED Do not change this file directly, use config.ts file instead
*
* @version 5
*/

import { AxiosRequestConfig } from \\"axios\\";
import { SwaggerResponse } from \\"./config\\";
import { Http } from \\"./httpRequest\\";
import { GetSettlementV1PrivateSettlementGetQueryParams, SettlementRequestResultListResponseVM,}  from \\"./types\\"

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const __DEV__ = process.env.NODE_ENV !== \\"production\\";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function overrideConfig(
  config?: AxiosRequestConfig,
  configOverride?: AxiosRequestConfig,
): AxiosRequestConfig {
  return {
    ...config,
    ...configOverride,
    headers: {
      ...config?.headers,
      ...configOverride?.headers,
    },
  };
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export function template(path: string, obj: { [x: string]: any } = {}) {
    Object.keys(obj).forEach((key) => {
      const re = new RegExp(\`{\${key}}\`, \\"i\\");
      path = path.replace(re, obj[key]);
    });

    return path;
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function objToForm(requestBody: { [name: string]: string | Blob | undefined }) {
  const formData = new FormData();

  Object.entries(requestBody).forEach(([key, value]) => {
    value && formData.append(key, value);
  });

  return formData;
}

export const getSettlementV1PrivateSettlementGet = (
    queryParams: GetSettlementV1PrivateSettlementGetQueryParams,configOverride?:AxiosRequestConfig
): Promise<SwaggerResponse< & (SettlementRequestResultListResponseVM)>> => {
  
  return Http.getRequest(
    getSettlementV1PrivateSettlementGet.key,
    queryParams,
    undefined,
    _CONSTANT1,
    overrideConfig(_CONSTANT0,
      configOverride,
    )
  )
}

/** Key is end point string without base url */
getSettlementV1PrivateSettlementGet.key = \\"/settlement/v1/private/settlement/get\\";
export const _CONSTANT0 = {
              headers: {
                \\"Content-Type\\": \\"application/json\\",
                Accept: \\"text/plain\\",
              },
            };export const _CONSTANT1 = [{\\"bearer\\":[]}];"
`;

exports[`generate generate hooks 1`] = `""`;

exports[`generate generate type 1`] = `
"
/**
* AUTO_GENERATED Do not change this file directly, use config.ts file instead
*
* @version 5
*/

        
        export interface GetSettlementV1PrivateSettlementGetQueryParams {\\"EndDate\\"?: string;\\"IsAuto\\"?: boolean;
/**
 * 
 * - Format: decimal
 */
\\"MinimumAmount\\"?: number;
/**
 * 
 * - Format: int32
 */
\\"Skip\\"?: number;
/**
 * 
 * - Format: date-time
 */
\\"StartDate\\"?: string;\\"States\\"?:  & (SettlementRequestStatus)[];
/**
 * 
 * - Format: int64
 */
\\"UserWalletId\\"?: number;\\"status\\"?: SettlementRequestStatus;}
        
        
        export type NotificationLevel =\\"Unknown\\" | \\"Default\\" | \\"Success\\" | \\"Info\\" | \\"Warning\\" | \\"Danger\\";
        
        
        export interface SettlementRequestResultListResponseVM {
/**
 * 
 * - Format: int64
 */
\\"totalCount\\": number;\\"settlementRequests\\"?: SettlementRequestResultResponseVM[];}
        
        
        export interface SettlementRequestResultResponseVM {\\"automaticSettlement\\": boolean;
/**
 * 
 * - Format: decimal
 */
\\"commissionAmount\\": number;
/**
 * 
 * - Format: date-time
 */
\\"createDateUtc\\": string;
/**
 * 
 * - Format: guid
 */
\\"creatorUserId\\": string;
/**
 * 
 * - Format: int32
 */
\\"domainId\\": number;
/**
 * 
 * - Format: int64
 */
\\"id\\": number;\\"status\\": SettlementRequestStatus;
/**
 * 
 * - Format: int64
 */
\\"userWalletId\\": number;\\"accountNumber\\"?: string;}
        
        
        export type SettlementRequestStatus =\\"Pending\\" | \\"Paid\\";
        "
`;
